### PR TITLE
Mosip 29780 develop

### PR DESCRIPTION
**Pull request by sowmya695**

### CODE REVIEW

File: registration-processor/core-processor/registration-processor-biometric-extraction-stage/src/test/java/io/mosip/registration/processor/stages/biometric/extraction/stages/test/BiometricExtractionStageTest.java [hunk 1]
```diff
@@ -331,4 +334,26 @@ public void testBiometricExtractionDraftException() throws Exception {
...
+		ErrorDTO errorDTO = new ErrorDTO();
+		errorDTO.setMessage("Failed to either encrypt/decrypt message using Kernel Crypto Manager");
+		errorDTO.setErrorCode("IDR-IDS-003");
+		IdResponseDTO idResponseDTO1 = new IdResponseDTO();
+		idResponseDTO1.setErrors(Lists.newArrayList(errorDTO));
+		when(registrationProcessorRestClientService.putApi(any(), any(), anyString(), anyString(), any(), any(), any()))
+				.thenReturn(idResponseDTO1);
+		MessageDTO result = biometricExtractionStage.process(messageDTO);
+		assertTrue(result.getInternalError());
+		assertTrue(result.getIsValid());
```
Comment (vishwa-vyom) [inline registration-processor/core-processor/registration-processor-biometric-extraction-stage/src/test/java/io/mosip/registration/processor/stages/biometric/extraction/stages/test/BiometricExtractionStageTest.java:357]: Should we not expect this as false ?

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/status/util/StatusUtil.java [hunk 2]
```diff
@@ -251,18 +251,33 @@ public enum StatusUtil {
...
 	FINALIZATION_FAILURE(StatusConstants.FINALIZATION_MODULE_FAILED + "001", "Draft request failed to publish."),
-	FINALIZATION_DRAFT_REQUEST_UNAVAILABLE(StatusConstants.FINALIZATION_MODULE_FAILED + "001",
+	FINALIZATION_DRAFT_REQUEST_UNAVAILABLE(StatusConstants.FINALIZATION_MODULE_FAILED + "002",
 			"Draft request is unavaialble in id-repo draft repository."),
 
+	FINALIZATION_IDREPO_DRAFT_EXCEPTION(StatusConstants.FINALIZATION_MODULE_FAILED + "003",
+			"Exception occured updating idrepo draft."),
+
+	FINALIZATION_IDREPO_DRAFT_REPROCESSABLE_EXCEPTION(StatusConstants.FINALIZATION_MODULE_FAILED + "004",
+			"Exception occured updating idrepo draft,Which can be reprocess"),
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/status/util/StatusUtil.java]: Can we change this to "Exception occured updating idrepo draft, which can be reprocessed". Similar for tha above messages also

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-biometric-extraction-stage/src/test/java/io/mosip/registration/processor/stages/biometric/extraction/stages/test/BiometricExtractionStageTest.java (hunk 1 above)
Comment (sowmya695) [reply registration-processor/core-processor/registration-processor-biometric-extraction-stage/src/test/java/io/mosip/registration/processor/stages/biometric/extraction/stages/test/BiometricExtractionStageTest.java:357]: In the existing code of BiometricExtractorStage we have code like if (registrationStatusDto.getLatestTransactionStatusCode()
				.equalsIgnoreCase(RegistrationTransactionStatusCode.REPROCESS.toString())) {
			object.setIsValid(true);
		}
So in test case i have added is valid true and in camel route also i have seen 
 <when>
            <jsonpath>$.[?(@['isValid'] == true &amp;&amp; @['internalError'] == true)]</jsonpath>
            <to uri="workflow-cmd://mark-as-reprocess" />
         </when>
So this test is fine

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/status/util/StatusUtil.java (hunk 2 above)
Comment (sowmya695) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/status/util/StatusUtil.java]: Fixed

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**